src/couch_file.erl

```diff
@@ -524,6 +525,99 @@ handle_info({'EXIT', _, Reason}, Fd) ->
     {stop, Reason, Fd}.
 
 
+%% first try to read header at end, falling back to chunked reads on failure
+find_last_header(Fd, FileSize) ->
+    ok = file:advise(Fd, 0, FileSize, dont_need), % minimize disk cache pollution
```

Comment:
I agree that calling it after use on a single chunk makes sense. FTR, http://man7.org/linux/man-pages/man2/posix_fadvise.2.html adds an additional note:

> Requests to discard partial pages are ignored.  It is preferable to
> preserve needed data than discard unneeded data.  If the application
> requires that data be considered for discarding, then offset and len
> must be page-aligned.

I don't the notes on dirty pages are significant to this case, I expect it's just pointing out the advisory nature of the call and the fact that unbacked storage won't be flushed by this call (vs. the reasonable expectation that it _does_ flush and obeys the advice).